patches/nginx-1.11.2-hook_error_log.patch

```diff
@@ -0,0 +1,71 @@
+diff --git a/src/core/ngx_cycle.h b/src/core/ngx_cycle.h
+index c51b7ff..765a0c1 100644
+--- a/src/core/ngx_cycle.h
++++ b/src/core/ngx_cycle.h
+@@ -25,6 +25,8 @@
+ typedef struct ngx_shm_zone_s  ngx_shm_zone_t;
+ 
+ typedef ngx_int_t (*ngx_shm_zone_init_pt) (ngx_shm_zone_t *zone, void *data);
++typedef ngx_int_t (*ngx_log_intercept_pt) (ngx_log_t *log, ngx_uint_t level,
++    u_char *buf, size_t len);
+ 
+ struct ngx_shm_zone_s {
+     void                     *data;
+@@ -75,6 +77,7 @@ struct ngx_cycle_s {
+     ngx_str_t                 prefix;
+     ngx_str_t                 lock_file;
+     ngx_str_t                 hostname;
++    ngx_log_intercept_pt      log_intercept_handler;
```

Comment:
Also, we should introduce a special macro like `HAVE_INTERCEPT_ERROR_LOG_PATCH`, similar to our existing `HAVE_ALLOW_REQUEST_BODY_UPDATING_PATCH` macro.